app/controllers/deposits_controller.rb

```diff
@@ -63,6 +63,58 @@ def destroy
     end
   end
 
+  def get_endpoint(collection_slug,
+                   depositor_user_id)
+    case COLLECTIONS[:slug][collection_slug][:parser]
+    when "academic-commons"
+      Sword::Endpoints::AcademicCommonsEndpoint.new(collection_slug, depositor_user_id)
+    when "proquest"
+      Sword::Endpoints::ProquestEndpoint.new(collection_slug, depositor_user_id)
+    when "eprints"
+      Sword::Endpoints::EprintsEndpoint.new(collection_slug, depositor_user_id)
+    else
+      # raise an exception here
+    end
+  end
+
+  def resubmit
+    original_deposit = Deposit.find(params[:id])
+    endpoint = get_endpoint(original_deposit.collection_slug,
+                            original_deposit.depositor_user_id)
+
+    path_to_deposit_contents = original_deposit.content_path
+
+    # log basic essential info. Keep it terse! Gonna use :warn level, though not a warning.
+    Rails.logger.warn("About to resubmit deposit for deposit id: #{original_deposit.id}:" \
```

Comment:
just wanted to mention how much i appreciate your approach to logging in this app. the thorough logs have been a huge help in troubleshooting self-deposit issues in the past. thanks!